src/sqlfluff/core/linter/linter.py

```diff
@@ -556,6 +585,21 @@ def lint_parsed(
             # If no parsed tree, set to None
             tree = None
             ignore_buff = []
+            if not parsed.config.get("disable_noqa"):
+                # Templating and/or parsing have failed. Look for "noqa"
+                # comments (the normal path for identifying these comments
+                # requires access to the parse tree, and because of the failure,
+                # we don't have a parse tree).
+                ignore_buff, ignore_violations = cls.extract_ignore_mask_source(
+                    parsed.source_str,
+                    [
+                        lm
+                        for lm in parsed.config.get("dialect_obj").lexer_matchers
+                        if lm.name == "inline_comment"
+                    ][0],
+                    [r.code for r in rule_set],
```

Comment:
At no point do we look for "noqa" here. Should we be?

Reply:
We don't need to. The new `extract_ignore_mask_source()` (called at line 593 above) looks for the `noqa` comments.